app/src/main/java/com/example/weatherapp/view/viewactivities/DetailsActivity.kt

```diff
@@ -24,7 +25,32 @@ class DetailsActivity: AppCompatActivity() {
 
     private val city: String by lazy { intent.getStringExtra("Temperature_Info").toString() }
 
-    private val detailsViewModel: WeatherDetailsViewModel by viewModels()
+    private val detailsViewModel: WeatherDetailsViewModel by viewModels() { DetailsViewModelFactory(city) }
+
+    override fun onCreate(savedInstanceState: Bundle?){
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_details_page)
+        supportActionBar?.hide()
+
+        lifecycleScope.launch {
+            launchTemperatureCall()
+            updateUiStateDetailsComponents()
+        }
+
+        returnButton.setOnClickListener{
+            finish()
+        }
+    }
+
+    private suspend fun updateUiStateDetailsComponents(){
+        detailsViewModel.uiState.collect { uiState ->
+            setComponentInfo(uiState)
+        }
+    }
+
+    private suspend fun launchTemperatureCall(){
+        detailsViewModel.requestTemperatureDetails()
```

Comment:
When is this call being made? What happens at configuration change? What if we wanted to preserve the data whenever a configuration change so we waste resources everytime the user rotates their phone?

Reply:
Now I understand that this call occurs each time the create method runs
After a screen rotation the onCreate method does run again. Therefore the network call has been moved to the init function of the viewModel. This is because when the activity dies or is restarted, the viewModel remains strong and lives on 💪